client/lib/paths/login/index.js

```diff
@@ -39,6 +39,7 @@ export function login( {
 	from = undefined,
 	allowSiteConnection = undefined,
 	signupUrl = undefined,
+	useQRCode = undefined,
```

Comment:
I might be missing something, but it's really weird to me that the default value for every argument in this function definition is "undefined." I'd think that be the _normal_ default value :P

Reply:
Yes I think it is really strange as well. I was following the convention that was already there. 
I think this was first done in https://github.com/Automattic/wp-calypso/pull/50962 

@sirbrillig for your thoughts on this :)

Reply:
When you call a JS function in a TS file, if the JS function has optional arguments, they need to have some way to tell the TS compiler that they are optional or they will be considered required. In this case, all these arguments are optional and so they have to have a default value or it will cause type errors. A better solution would be to convert the function to TS but that was outside the scope of the above-mentioned PR.

Reply:
Thanks for the explanation @sirbrillig 

Maybe we can rewrite that function in a different PR. 
I really think it could be a lot simpler. 🤞  - famous last words.